src/main/java/Parser.java

```diff
@@ -0,0 +1,35 @@
+public class Parser {
+    public static Command parse(String fullCommand) throws DukeException{
+        fullCommand = fullCommand.trim();
+        int spacePosition = fullCommand.indexOf(" ");
+        if(spacePosition == -1 && !(fullCommand.equals("list") || fullCommand.equals("bye"))){
+            raiseSingleBadInput(fullCommand);
+        }
+        String[] taskInfo = fullCommand.split(" ", 2);
+        switch(taskInfo[0]){
+            case "todo":
```

Comment:
Interesting way of handling the different cases! I wouldn't have thought of doing this. However, according to the style guide for CS2103, you should comment that you are allowing a fall-through case. I feel that this makes it clearer to a less-experienced programmer as well. (: